factory.py

```diff
@@ -0,0 +1,41 @@
+import functools
+from typing import Callable
+
+
+def type_check(correct_type: any) -> Callable:
```

Comment:
You are using a function instead of the actual type. You should import the matching type from typing module  - 
https://docs.python.org/3/library/typing.html
any is a function that does another thing